Sources/FluentKit/Properties/Parent.swift

```diff
@@ -1,69 +1,120 @@
+extension Parent {
+    internal enum EagerLoaded: CustomStringConvertible {
+        case loaded(To)
+        case notLoaded
 
-    public func get(on database: Database) -> EventLoopFuture<To> {
-        return self.query(on: database).first().flatMapThrowing { parent in
-            guard let parent = parent else {
-                throw FluentError.missingParent
+        var description: String {
+            switch self {
+            case let .loaded(model): return model.description
+            case .notLoaded: return "notLoaded"
             }
-            return parent
         }
     }
-
 }
 
 extension Parent: FieldRepresentable {
-    public var field: Field<To.IDValue> {
-        return self.$id
-    }
+    public var field: Field<To.IDValue> { self.$id }
 }
 
 extension Parent: AnyProperty {
     func encode(to encoder: Encoder) throws {
         var container = encoder.singleValueContainer()
-        if let parent = self.eagerLoadedValue {
+
+        if case let .loaded(parent) = self.eagerLoadedValue {
             try container.encode(parent)
         } else {
-            try container.encode([
-                To.key(for: \._$id): self.id
-            ])
+            try container.encode([self.$id.key: self.id])
         }
     }
 
     func decode(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: _ModelCodingKey.self)
-        try self.$id.decode(from: container.superDecoder(forKey: .string(To.key(for: \._$id))))
+        try self.$id.decode(from: container.superDecoder(forKey: .string(self.$id.key)))
         // TODO: allow for nested decoding
     }
 }
 
 extension Parent: AnyField { }
+
+
+// MARK: - Specialization
+
+extension Parent where To: Model {
+    public convenience init(key: String) {
+        self.init(id: key, eagerLoadRequest: SubqueryEagerLoad(key: key))
+    }
+
+    public func query(on database: Database) -> QueryBuilder<To> {
+        return To.query(on: database).filter(\._$id == self.id)
+    }
+
+    public func get(on database: Database) -> EventLoopFuture<To> {
+        return self.query(on: database).first().flatMapThrowing { parent in
+            guard let parent = parent else {
+                throw FluentError.missingParent
+            }
+            return parent
+        }
+    }
+}
+
+extension Parent where To: OptionalType, To.Wrapped: Model, To.IDValue == To.Wrapped.IDValue? {
+    public convenience init(key: String) {
+        self.init(id: key, eagerLoadRequest: SubqueryEagerLoad(key: key))
+    }
+
+    public func query(on database: Database) -> QueryBuilder<To.Wrapped> {
+        return To.Wrapped.query(on: database).filter(\To.Wrapped._$id == self.id)
+    }
+
+    public func get(on database: Database) -> EventLoopFuture<To> {
+        return self.query(on: database).first().flatMapThrowing { parent in parent as! To }
+    }
+}
+
+
+// MARK: - Optionals
+
+public typealias GenericModel = ModelIdentifiable & Codable & CustomStringConvertible
```

Comment:
Maybe we call this `ModelOrOptional` or `OptionalModel` to be clear?

Reply:
I think making it an explicit protocol could be nice, too since realistically we only want `Model` and `Optional<Model>` to conform.